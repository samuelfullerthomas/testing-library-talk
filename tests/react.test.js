// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, waitFor, screen } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
import axiosMock from "axios";
// the component to test
import Fetch from "./fetch";

// https://jestjs.io/docs/en/mock-functions#mocking-modules
jest.mock("axios");

test("loads and displays greeting", async () => {
  // Arrange
  const url = "/greeting";
  const { asFragment } = render(<Fetch url={url} />);

  axiosMock.get.mockResolvedValueOnce({
    data: { greeting: "hello there" },
  });

  // Act
  fireEvent.click(screen.getByText("Load Greeting"));
  // Wait until the mocked `get` request promise resolves and
  // the component calls setState and re-renders.
  // `waitFor` waits until the callback doesn't throw an error
  await waitFor(() => screen.getByRole("heading"));

  // Assert
  expect(axiosMock.get).toHaveBeenCalledTimes(1);
  expect(axiosMock.get).toHaveBeenCalledWith(url);
  expect(screen.getByRole("heading")).toHaveTextContent("hello there");
  expect(screen.getByRole("button")).toHaveAttribute("disabled");

  expect(asFragment()).toMatchSnapshot();
});
