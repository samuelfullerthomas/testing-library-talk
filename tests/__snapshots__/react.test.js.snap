// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays greeting 1`] = `
<DocumentFragment>
  <div>
    <button
      disabled=""
    >
      Ok
    </button>
    <h1>
      hello there
    </h1>
  </div>
</DocumentFragment>
`;
